lib/sla.js

```diff
@@ -183,37 +265,37 @@ function CustomError(name, message) {
 /**
  * Receives initial input and begins the application.
  * @param {Object} client - Client received from app.js.
+ * @param {string} confFile - the configuration path and file name
  * @param {Object} inbound - the inbound channel
- * @param {String} bridgeName - the name of the bridge to be used/created
+ * @param {string} extension - the name of the extension to access
  * @return {Q} Q promise object.
  */
-module.exports = function(client, inbound, bridgeName) {
-  //This specifies what bridge number to use for SLA
-  if(checkIfNum(bridgeName)) {
-    var answer = Q.denodeify(inbound.answer.bind(inbound));
-    return answer()
-      .then(function() {
-        return designateMixingBridge(client, bridgeName);
-      }).then(function (bridge) {
-        return originateChannel(client, inbound, bridge)
-          .then(function (dialed) {
-            // If dialed is an actual channel, it will not be a string
-            if (typeof dialed !== 'string') {
-              return addChannelsToBridge(inbound, dialed, bridge);
-            }
-            else {
-              var hangup = Q.denodeify(inbound.hangup.bind(inbound));
-              hangup();
-              throw new CustomError('EarlyHangup',
-                dialed);
-            }
-          });
-      });
-  }
-  else {
-    var hangup = Q.denodeify(inbound.hangup.bind(inbound));
-    hangup();
-    return Q.reject(new CustomError('SLASpecficationError',
-          'Not a numeric SLA specification: ' + bridgeName));
-  }
+module.exports = function(client, confFile, inbound, extension) {
+  return getTrunkData(confFile, extension)
+    .then(function(trunk) {
+      var answer = Q.denodeify(inbound.answer.bind(inbound));
+      return answer()
+        .then(function() {
+          return designateMixingBridge(client, extension);
+        }).then(function (bridge) {
+          return originateChannel(client, inbound, bridge, trunk)
+            .then(function (dialed) {
+              // If dialed is an actual channel, it will not be a string
+              if (typeof dialed !== 'string') {
+                return addChannelsToBridge(inbound, dialed, bridge);
+              }
+              else {
+                throw new CustomError('EarlyHangup',
+                  dialed);
+              }
+            });
+        });
+    })
+  .catch(function (err) {
+    if(err.message !== 'Inbound channel hungup') {
```

Comment:
You should use a custom error and check err.name instead. Error messages can change, error names should not.